artifact/signer.go

```diff
@@ -300,5 +298,33 @@ func GetKeyAndSignMethod(keyPEM []byte) (*SigningMethod, error) {
 		}
 		return &SigningMethod{Key: ecdsaKey, Public: pub, Method: new(ECDSA256)}, nil
 	}
-	return nil, errors.Wrap(err, "signer: unsupported private key type or error occured")
+
+	key, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+	if _, ok := key.(*ecdsa.PrivateKey); ok {
+		ecdsaKey := key.(*rsa.PrivateKey)
```

Comment:
The `ok` and the actual casting are opposite types. Typo?

It might be better to assign both variables in the first cast. Then the problem cannot happen in the first place.

Reply:
damn copy-paste after last force push when I swapped the identifiers. fixed.

Reply:
it was a complete mixup, methods and even log messages were inter-mixed. should be ok now.

Reply:
and thanks for catching.